splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsOperation.java

```diff
@@ -93,8 +93,13 @@ public StatisticsOperation(ScanSetBuilder scanSetBuilder, boolean useSample, dou
                     statsDataSet = dsp.readTextFile(null, builder.getLocation(), builder.getDelimited(), null, builder.getColumnPositionMap(), null, null, null, builder.getTemplate(), useSample, sampleFraction);
                 else if (storedAs.equals("P"))
                     statsDataSet = dsp.readParquetFile(builder.getColumnPositionMap(), builder.getPartitionByColumnMap() , builder.getLocation(), null, null, null, builder.getTemplate(), useSample, sampleFraction);
-                else if (storedAs.equals("O"))
-                    statsDataSet = dsp.readORCFile(builder.getColumnPositionMap(), builder.getPartitionByColumnMap(), builder.getLocation(), null, null, null, builder.getTemplate(), useSample, sampleFraction);
+                else if (storedAs.equals("O")) {
+                    int[] baseColumnMap = builder.getColumnPositionMap();
+                    for (int i = 0; i < baseColumnMap.length; i++) {
+                        baseColumnMap[i] = baseColumnMap[i] - 1;  //make it zero based
```

Comment:
There are places in code that use (baseColumnMap[I] == -1) to check if it is a valid column id or to terminate a loop over the baseColumnMap array. Unconditionally subtracting every element by 1 may cause confusion as we will see -2 in the cells.  Could you change the logic to only update "if (baseColumnMap[I] >0)".